drivers/shared/executor/executor_linux.go

```diff
@@ -805,52 +788,125 @@ func cmdMounts(mounts []*drivers.MountConfig) []*lconfigs.Mount {
 	return r
 }
 
-// lookupTaskBin finds the file `bin` in taskDir/local, taskDir in that order, then performs
-// a PATH search inside taskDir. It returns an absolute path. See also executor.lookupBin
-func lookupTaskBin(command *ExecCommand) (string, error) {
+// lookupTaskBin finds the file `bin` in taskDir/local, taskDir in that order,
+// then performs a PATH search inside taskDir. It returns an absolute path
+// inside the container that will get passed as arg[0] to the launched process,
+// and the absolute path to that binary as seen by the host (these will be
+// identical for binaries that don't come from mounts).
+//
+// See also executor.lookupBin for a version used by non-isolated drivers.
+func lookupTaskBin(command *ExecCommand) (string, string, error) {
 	taskDir := command.TaskDir
 	bin := command.Cmd
```

Comment:
does it make sense to search `task/local/` and `task/` if the `bin` is given as absolute? don't think i've ever tried that, but it sounds confusing if it works

Reply:
It _does_ sound confusing, and I spent much of yesterday puzzled over that! But it's the existing behavior (ref [`executor_linux.go#L814-L819`](https://github.com/hashicorp/nomad/blob/main/drivers/shared/executor/executor_linux.go#L814-L819)) so changing it now will break backwards compatibility. Let's save changing it for `exec` v2.